src/xcode/ENA/ENA/Source/HTTPClient/Resources/TraceWarningDiscovery/TraceWarningDiscoveryResource.swift

```diff
@@ -0,0 +1,68 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import Foundation
+
+struct TraceWarningDiscoveryResource: Resource {
+
+	// MARK: - Init
+	init(
+		unencrypted: Bool,
+		country: String,
+		isFake: Bool = false,
+		trustEvaluation: TrustEvaluating = DefaultTrustEvaluation(
+			publicKeyHash: Environments().currentEnvironment().pinningKeyHashData
+		)
+	) {
+		self.locator = .traceWarningDiscovery(unencrypted: unencrypted, country: country, isFake: isFake)
+		self.type = .default
```

Comment:
Shouldn't it be caching? Because I see something with ETag in the old code.

Reply:
no the eTag is only used inside the datastorage - the api it self doesn't do any eTag logic here. Strange but true.